hermes/src/network/rpc/ws.rs

```diff
@@ -0,0 +1,290 @@
+                    .clone();
+                let price_feed =
+                    RpcPriceFeed::from_price_info(price_info, config.verbose, config.binary);
+                // Feed does not flush the message and will allow us
+                // to send multiple messages in a single flush.
+                self.sender
+                    .feed(Message::Text(
+                        serde_json::to_string(&ServerMessage::PriceUpdate { price_feed }).unwrap(),
+                    ))
+                    .await?;
+            }
+        }
+        self.sender.flush().await?;
+        Ok(())
+    }
+
+    async fn handle_client_message(&mut self, message: Message) -> Result<()> {
+        if let Message::Close(_) = message {
+            log::debug!("Subscriber {} closed connection", self.id);
+            self.closed = true;
+            return Ok(());
+        }
+
+        let maybe_client_message = match message {
+            Message::Text(text) => serde_json::from_str::<ClientMessage>(&text),
+            Message::Binary(data) => serde_json::from_slice::<ClientMessage>(&data),
+            _ => {
+                return Ok(());
+            }
+        };
+
+        match maybe_client_message {
+            Err(e) => {
+                self.sender
+                    .feed(
+                        serde_json::to_string(&ServerMessage::Response(
+                            ServerResponseMessage::Err {
+                                error: e.to_string(),
+                            },
+                        ))
+                        .unwrap()
+                        .into(),
+                    )
+                    .await?;
+                return Ok(());
+            }
+            Ok(ClientMessage::Subscribe {
+                ids,
+                verbose,
+                binary,
+            }) => {
+                for id in ids {
+                    let price_id: PriceIdentifier = id.into();
+                    self.price_feeds_with_config
+                        .insert(price_id, PriceFeedClientConfig { verbose, binary });
+                }
+            }
+            Ok(ClientMessage::Unsubscribe { ids }) => {
+                for id in ids {
+                    let price_id: PriceIdentifier = id.into();
+                    self.price_feeds_with_config.remove(&price_id);
+                }
+            }
+        }
+
+        self.sender
+            .send(
+                serde_json::to_string(&ServerMessage::Response(ServerResponseMessage::Ok))
+                    .unwrap()
+                    .into(),
+            )
+            .await?;
+
+        Ok(())
+    }
+}
+
+pub async fn dispatch_updates(update_feed_ids: Vec<PriceIdentifier>, state: super::State) {
+    let ws_state = state.ws.clone();
+    let mut closed_subscribers: Vec<SubscriberId> = Vec::new();
+
+    for subscriber in ws_state.subscribers.iter_mut() {
+        if let Err(e) = subscriber.send(update_feed_ids.clone()).await {
```

Comment:
I think you want these sends to run in parallel right? 

consider doing a join! like this https://rust-lang.github.io/async-book/06_multiple_futures/02_join.html

Reply:
actually you probably need this https://docs.rs/futures/0.3.5/futures/future/fn.join_all.html

Reply:
Oh yeah I totally forgot it. I'll do it.